src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -1,242 +1,54 @@
-using System.Collections.Generic;
+using System;
 using System.IO;
-using System.Linq;
 using System.Text;
 using GitVersion.Configuration.Init.Wizard;
-using GitVersion.VersioningModes;
-using GitVersion.Extensions;
-using GitVersion.Common;
 using GitVersion.Logging;
 
 namespace GitVersion.Configuration
 {
-    public class ConfigurationProvider
+    public class ConfigurationProvider : IConfigurationProvider
```

Comment:
We should be consistently using `Config` or `Configuration`, not both. Since we use `Configuration` in the namespace, perhaps `Config` is OK? Alternatively `GitVersionConfiguration` to avoid collision.

Reply:
Renamed to `ConfigProvider`, using `ConfigExtensions`